### ISSUE TITLE
edn-style records syntax

### ISSUE THREAD

**Issue by jjl** [jjl/lfe#5]
I think we should support records with the edn-style syntax as a nice shorthand:

```
#foo{bar=baz} ; instantiate record foo with bar=baz
```

I'd also like to be able to use the elixir records-as-maps thing

```
#foo:elixir{bar=baz}
```

(where we could optionally expand later to support any others). I haven't yet found a compelling alternative use for the colon here.

**Comment by yurrriq** [comment]
That's Erlang syntax too :)

**Comment by jjl** [comment]
Yeah, but shush or robert will make sure i never hear the end of it ;)
